aleph-client/src/lib.rs

```diff
@@ -122,6 +124,27 @@ pub fn send_xt(connection: &Connection, xt: String, xt_name: &'static str, tx_st
     );
 }
 
+pub fn try_send_xt(
+    connection: &Connection,
+    xt: String,
+    xt_name: &'static str,
+    tx_status: XtStatus,
+) -> ApiResult<()> {
+    let block_hash = connection
+        .send_extrinsic(xt, tx_status)?
+        .ok_or_else(|| Error::Other(String::from("Could not get block hash").into()))?;
+    let block_number = connection
+        .get_header::<Header>(Some(block_hash))?
+        .expect("Block exists; qed")
+        .number;
+    info!(
```

Comment:
not neccessarily, what if you just dispatch into mempool and do not wait for receipt ?

Reply:
yeah, the original method has some defects, but I wanted to keep it consistent :|

Reply:
I guess I should 'fix' both methods

Reply:
done, (narrowed to only some variants)